cmake/public/gridtools_setup_targets.cmake

```diff
@@ -219,27 +219,13 @@ macro(_gt_setup_targets _config_mode clang_cuda_mode)
     target_link_libraries(${_gt_namespace}stencil_naive INTERFACE ${_gt_namespace}gridtools)
 
     set(_required_nlohmann_json_version "3.7.3")
-    if(NOT _nlohmann_json_already_fetched)
-        find_package(nlohmann_json ${_required_nlohmann_json_version})
-    endif()
-    if(NOT nlohmann_json_FOUND)
-        set(_gridtools_repository "https://github.com/GridTools/gridtools.git")
-        set(_gridtools_tag        "v${_required_gridtools_version}")
-        include(FetchContent)
-        FetchContent_Declare(json
-                GIT_REPOSITORY https://github.com/nlohmann/json.git
-                GIT_TAG "v${_required_nlohmann_json_version}")
-        FetchContent_GetProperties(json)
-        if(NOT json_POPULATED)
-            FetchContent_Populate(json)
-            set(JSON_BuildTests OFF CACHE INTERNAL "")
-            add_subdirectory(${json_SOURCE_DIR} ${json_BINARY_DIR} EXCLUDE_FROM_ALL)
-        endif()
-        set(_nlohmann_json_already_fetched ON CACHE INTERNAL "")
-    endif()
+    include(get_nlohmann_json)
+    get_nlohmann_json(${_required_nlohmann_json_version})
 
-    _gt_add_library(${_config_mode} stencil_dump)
-    target_link_libraries(${_gt_namespace}stencil_dump INTERFACE ${_gt_namespace}gridtools nlohmann_json::nlohmann_json)
+    if(TARGET nlohmann_json::nlohmann_json)
+        _gt_add_library(${_config_mode} stencil_dump)
```

Comment:
if we add the target conditionally, the usage of thus target in the `tests/regression/CMakeList.txt` should be guarded with `if(TARGET stencil_dump) ...`

Reply:
Thanks, I forgot that it's still conditional (but only if download fails).